kytos/core/interface.py

```diff
@@ -173,19 +177,295 @@ def status_reason(self):
+                    tag_ranges[0][1] + 1 >= available[0][0]):
+                reason = f"There is a gap at the beginning: available_tag:"\
+                         f"{available[0]} and tag_ranges:"\
+                         f"{tag_ranges[0]}. Tags can be {[1, 4095]}."
+                return reason
+        if available[-1][1] != 4095:
+            if not (tag_ranges[-1][1] == 4095 and
+                    tag_ranges[-1][0] <= available[-1][1] + 1):
+                reason = f"There is a gap at the end: available_tag:"\
+                         f"{available[-1]} and tag_ranges:"\
+                         f"{tag_ranges[-1]}. Tags can be {[1, 4095]}."
+                return reason
+        return None
+
+    def not_applicable(
+        self,
+        tag_ranges: list[list[int, int]],
+        tag_type: str = '1'
+    ) -> Optional[str]:
+        """Check if tag_ranges can be applied when there is available
+        Returns an early False when a gap is detected
+        Simulates sliding window O(n+m)"""
+        available = self.available_tags[tag_type]
+        if not available:
+            return "Every tag is used."
+
+        error_msg = self.verify_first_last_tag(tag_ranges)
+        if error_msg:
+            return error_msg
+
+        # Corner case
+        ava_n = len(available)
+        if ava_n == 1:
+            return None
+
+        rest_n = len(tag_ranges)
+        ava_i, rest_i = 0, 0
+        while rest_i < rest_n:
+            if tag_ranges[rest_i][0] <= available[ava_i][1] + 1:
+                while ava_i + 1 < ava_n:
+                    if tag_ranges[rest_i][1] + 1 >= available[ava_i+1][0]:
+                        ava_i += 1
+                    elif tag_ranges[rest_i][1] <= available[ava_i][1]:
+                        break
+                    else:
+                        reason = f"Gap detected available_tag:"\
+                                 f"{[available[ava_i], available[ava_i+1]]}"\
+                                 f" and tag_ranges:{tag_ranges[rest_i]}."
+                        return reason
+                rest_i += 1
+            else:
+                reason = f"Gap detected available_tag:"\
+                         f"{[available[ava_i], available[ava_i+1]]} and "\
+                         f"tag_ranges:{tag_ranges[rest_i]}."
+                return reason
+        return None
+
+    def set_tag_ranges(self, tag_ranges: list[list[int]], tag_type: str):
+        """Set new restriction"""
+        with self._tag_lock:
+            reason = self.not_applicable(tag_ranges, tag_type)
+            if reason:
+                msg = f"tag_ranges: {tag_ranges} is not applicable. {reason}"
+                raise KytosResizingAvailableTagError(msg)
+            self.tag_ranges[tag_type] = tag_ranges
+
+    def remove_tag_ranges(self, tag_type: str = '1'):
+        """Remove restriction set on tag_ranges"""
+        self.tag_ranges[tag_type] = None
+
+    def get_next_available_tag(self) -> Optional[int]:
```

Comment:
Why isn't the type being passed too?

Reply:
I missed this method, it is because this method is not used and it was meant to be deleted.

Reply:
Got it. Indeed, NApps are only getting the next available for links, since on UNI interfaces they won't want a whatever next available tag, let's nuke this. It's great how link `get_next_available_tag` is shaping up.